src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -234,13 +237,26 @@ private function appendToken(SessionInterface $sessionContainer, Response $respo
             return FigResponseCookies::set($response, $this->getExpirationCookie());
         }
 
-        if (! $sessionContainerChanged) {
-            return $response;
+        if ($sessionContainerChanged || $this->shouldTokenBeRefreshed($token)) {
+            return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        }
+
+        return $response;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    private function shouldTokenBeRefreshed(Token $token = null) : bool
+    {
+        if (null === $token) {
+            return false;
         }
 
-        return FigResponseCookies::set($response, $this->getTokenCookie($sessionContainer));
+        return time() >= ($token->getClaim('exp') - $this->refreshTime);
```

Comment:
As discussed with @malukenho, this expression is not fully true.

To make this more practical:

`<token-created>=========<token-refresh>=========================================<token-expires>`

The time between `token-created` and `token-expires` is `expirationTime`, but what do we name the time between `token-created` and `token-refresh`? 
Also, is this operation valid right now? Or are we just subtracting (by default) 60 seconds from the end?